router.go

```diff
@@ -52,16 +54,21 @@ type MapRouter struct {
 	serviceProcedures         map[serviceProcedure]transport.Procedure
 	serviceProcedureEncodings map[serviceProcedureEncoding]transport.Procedure
 	supportedEncodings        map[serviceProcedure][]string
+	serviceNames              map[string]struct{}
 }
 
 // NewMapRouter builds a new MapRouter that uses the given name as the
 // default service name.
 func NewMapRouter(defaultService string) MapRouter {
+	serviceNamesMap := map[string]struct{}{
```

Comment:
we can actually just inline this in the return
```go
return MapRouter{
  ...
    supportedEncodings:    make(map[serviceProcedure][]string)
    serviceNames:          map[string]struct{}{
      defaultService: {},
    }
}
```

Reply:
resolved.